From 253cce40677613b2462077b195b4230ebc87a3df Mon Sep 17 00:00:00 2001
From: jakub <jakub@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Tue, 24 Jun 2014 15:13:32 +0000
Subject: [PATCH 066/134] 	PR target/61570 	*
 config/i386/driver-i386.c (host_detect_local_cpu): For unknown 	model
 family 6 CPU with has_longmode never use a CPU without 	64-bit
 support.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_8-branch@211948 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog                 | 7 +++++++
 gcc/config/i386/driver-i386.c | 5 +++++
 2 files changed, 12 insertions(+)

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index 45da744..44896fb 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -1,3 +1,10 @@
+2014-06-24  Jakub Jelinek  <jakub@redhat.com>
+
+	PR target/61570
+	* config/i386/driver-i386.c (host_detect_local_cpu): For unknown
+	model family 6 CPU with has_longmode never use a CPU without
+	64-bit support.
+
 2014-06-20  Chung-Lin Tang  <cltang@codesourcery.com>
 
 	Backport from mainline
diff --git a/gcc/config/i386/driver-i386.c b/gcc/config/i386/driver-i386.c
index 148fbc2..87b6a1b 100644
--- a/gcc/config/i386/driver-i386.c
+++ b/gcc/config/i386/driver-i386.c
@@ -713,6 +713,11 @@ const char *host_detect_local_cpu (int argc, const char **argv)
 		    /* Assume Core 2.  */
 		    cpu = "core2";
 		}
+	      else if (has_longmode)
+		/* Perhaps some emulator?  Assume x86-64, otherwise gcc
+		   -march=native would be unusable for 64-bit compilations,
+		   as all the CPUs below are 32-bit only.  */
+		cpu = "x86-64";
 	      else if (has_sse3)
 		/* It is Core Duo.  */
 		cpu = "pentium-m";
-- 
1.8.5.5

